privacyidea/lib/smsprovider/ScriptSMSProvider.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+#
+#    2020-06-04 Cornelius Kölbel <cornelius.koelbel@netknights.i>
+#               Initial Code
+#
+#    This program is free software: you can redistribute it and/or
+#    modify it under the terms of the GNU Affero General Public
+#    License, version 3, as published by the Free Software Foundation.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the
+#               GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__doc__ = """This is the SMSClass to send SMS via a script.
+"""
+
+from privacyidea.lib.smsprovider.SMSProvider import (ISMSProvider, SMSError)
+from privacyidea.lib import _
+from privacyidea.lib.framework import get_app_config_value
+import subprocess
+import logging
+import traceback
+log = logging.getLogger(__name__)
+
+
+SCRIPT_BACKGROUND = "background"
+SCRIPT_WAIT = "wait"
+
+
+class ScriptSMSProvider(ISMSProvider):
+
+    def __init__(self, db_smsprovider_object=None, smsgateway=None, directory=None):
+        """
+        Create a new SMS Provider object fom a DB SMS provider object
+
+        :param db_smsprovider_object: The database object
+        :param smsgateway: The SMS gateway object from the database table
+            SMS gateway. The options can be accessed via
+            self.smsgateway.option_dict
+        :return: An SMS provider object
+        """
+        self.config = db_smsprovider_object or {}
+        self.smsgateway = smsgateway
+        self.script_directory = directory or get_app_config_value("PI_SCRIPT_SMSPROVIDER_DIRECTORY",
+                                                                  "/etc/privacyidea/scripts")
+
+    def submit_message(self, phone, message):
+        """
+        send a message to a phone via an smpp protocol to smsc
```

Comment:
We just call the script, we have no idea about the protocol, or have we?

Reply:
done in 8d67564e